internal/db/nosql/cassandra/cassandra.go

```diff
@@ -93,36 +80,55 @@ type client struct {
 		dcHost    string
 		whiteList []string
 	}
-	socketKeepalive          time.Duration
-	maxPreparedStmts         int
-	maxRoutingKeyInfo        int
-	pageSize                 int
-	serialConsistency        gocql.SerialConsistency
-	tls                      *tls.Config
-	tlsCertPath              string
-	tlsKeyPath               string
-	tlsCAPath                string
-	enableHostVerification   bool
-	defaultTimestamp         bool
-	reconnectInterval        time.Duration
-	maxWaitSchemaAgreement   time.Duration
-	ignorePeerAddr           bool
-	disableInitialHostLookup bool
-	disableNodeStatusEvents  bool
-	disableTopologyEvents    bool
-	disableSchemaEvents      bool
-	disableSkipMetadata      bool
-	queryObserver            QueryObserver
-	batchObserver            BatchObserver
-	connectObserver          ConnectObserver
-	frameHeaderObserver      FrameHeaderObserver
-	defaultIdempotence       bool
-	dialer                   gocql.Dialer
-	writeCoalesceWaitTime    time.Duration
+	client struct {
+		hosts                    []string
+		cqlVersion               string
+		protoVersion             int
+		timeout                  time.Duration
+		connectTimeout           time.Duration
+		port                     int
+		keyspace                 string
+		numConns                 int
+		consistency              gocql.Consistency
+		compressor               gocql.Compressor
+		username                 string
+		password                 string
+		authProvider             func(h *gocql.HostInfo) (gocql.Authenticator, error)
+		retryPolicy              retryPolicy
```

Comment:
Could you please give me a reason of this implementation?
there are no method of `retryPolicy ` and `reconnectionPolicy `, `poolConfig `, `hostFilter ` .
So in my opinion, I think inner struct is better.

Reply:
it is only for testing. I need to implement the custom comparer for these struct, like https://github.com/vdaas/vald/pull/638/files#diff-de534877b6350fd05c5419656dbb4242R42
, so if I don't define the struct here, the comparer cannot refer to this struct definition.